src/jokes.ts

```diff
@@ -60,6 +60,41 @@ class JokesLoader {
     }
   }
 
+  public async deleteJoke(joke: Joke): Promise<{ success: boolean; joke_id?: number; error?: string }> {
+    const jokesPath = path.join(__dirname, '../blagues.json');
+    const req = await this.checkAccess(jokesPath);
+
+    if (!req.success) return req;
+
+    try {
+      await this.loader.wait();
+
+      const rawData = await fs.readFile(jokesPath, 'utf-8');
+      const jokes = (rawData.length ? JSON.parse(rawData) : []) as Joke[];
+      const toDelJoke = jokes.find((j) => j.id === joke.id)!;
```

Comment:
Vaut mieux quand meme check si il y'a une blague ou pas au cas ou. 
```suggestion
      const toDelJoke = jokes.find((j) => j.id === joke.id);

      if (!toDelJoke) return { success: false, error: `La blague n'a pas été trouvée.` };

```